peps/pep-0763.rst

```diff
@@ -0,0 +1,275 @@
+standardized or subject to standardization, such as the web interface and
+signed-in user operations. As a result, this section describes its
+implementation in behavioral terms.
+
+Changes
+-------
+
+* Per the eligibility rules above, PyPI will reject web interface requests
+  (using an appropriate HTTP response code of its choosing) for
+  file, release, or project deletion if the respective object is not
+  eligible for deletion.
+* PyPI will amend its web interface to indicate a file/release/project's
+  deletion ineligibility, e.g. by styling the relevant UI elements as "inactive"
+  and making relevant bottoms/forms unclickable.
+
+Security Implications
+=====================
+
+This PEP does not identify any positive or negative security implications
+associated with proposed approach.
+
+How To Teach This
+=================
+
+This PEP suggests at least two pieces of public-facing material to help
+the larger Python packaging community (and its downstream consumers)
+understand its changes:
+
+* An announcement post on the `PyPI blog <https://blog.pypi.org>`_ explaining
+  the nature of the PEP and its behavioral implications for PyPI.
+* Updates to the `PyPI user documentation <https://docs.pypi.org/>`_ explaining
+  the difference between deletion and yanking and the limited conditions under
+  which the former can still be initiated by package owners.
+
+Rejected Ideas
+==============
+
+Conditioning deletion on dependency relationships
+-------------------------------------------------
+
+An alternative to time-based deletion windows is deletion eligibility based on
+downstream dependents. For example, a release could be considered deletable
+if and only if it has fewer than ``N`` downstream dependents on PyPI,
+where ``N`` could be as low as 1.
+
+This idea is appealing since it directly links deletion eligibility to
+disruptiveness. `NPM <https://www.npmjs.com/>`_ uses it and
+conditions project removal on the absence of any downstream dependencies
+known to the index.
+
+Despite its appeal, this PEP identifies several disadvantages and technical
+limitations that make dependency-conditioned deletion not appropriate
+for PyPI:
+
+1. *PyPI is not aware of dependency relationships.* In Python packaging,
+   both project builds *and* metadata generation are frequently dynamic
+   operations, involving arbitrary project-specified code. This is typified
+   by source distributions containing ``setup.py`` scripts, where the execution
+   of ``setup.py`` is responsible for computing the set of dependencies
+   encoded in the project's metadata.
+
+   This is in marked contrast to ecosystems like NPM and Rust's
+   `crates <https://crates.io/>`_, where project *builds* can be dynamic but
+   the project's metadata itself is static.
+
+   As a result of this,
+   `PyPI doesn't know your project's dependencies <https://dustingram.com/articles/2018/03/05/why-pypi-doesnt-know-dependencies/>`_,
```

Comment:
FYI -- you can split these links over multiple lines, if you'd like, the following should work:

```suggestion
   As a result of this, `PyPI doesn't know your project's dependencies
   <https://dustingram.com/articles/2018/03/05/why-pypi-doesnt-know-dependencies/>`_,
```

Reply:
TIL, thanks!